bio_nautilus.py

```diff
@@ -0,0 +1,294 @@
+            index: Integer position or slice object.
+
+        Returns:
+            Single character (str) for index, or subsequence (BiologicalSequence) for slice.
+        """
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        """Return the sequence as a plain string."""
+        pass
+
+    @abstractmethod
+    def __repr__(self):
+        """Return an unambiguous string representation of the object."""
+        pass
+
+    @abstractmethod
+    def check_alphabet(self):
+        """Check if all characters in the sequence are valid."""
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence, ABC):
+    """Abstract class representing a nucleic acid sequence (DNA/RNA)."""
+
+    _valid_chars = None
+    _complement_map = None
+
+    def __init__(self, sequence):
+        """
+        Initialize a nucleic acid sequence.
+
+        Args:
+            sequence: Input sequence string.
+
+        Raises:
+            ValueError: If invalid characters are present.
+            NotImplementedError: if NucleicAcidSequence is called directly.
+        """
+        if self.__class__ == NucleicAcidSequence:
+            raise NotImplementedError(
+                "NucleicAcidSequence является абстрактным классом и не может быть создан напрямую."
+            )
+        self.sequence = sequence.upper()
+        if not self.check_alphabet():
+            raise ValueError("Недопустимые символы нуклеотидов")
+
+    def check_alphabet(self):
+        """
+        Verify that all characters are in the allowed alphabet.
+
+        Returns:
+            True if all characters are valid, False otherwise.
+
+        Raises:
+            NotImplementedError: If _valid_chars is not defined in subclass.
+        """
+        if self._valid_chars is None:
+            raise NotImplementedError("Допустимые символы не определены")
+        return all(c in self._valid_chars for c in self.sequence)
+
+    def complement(self):
+        """
+        Create the complementary sequence.
+
+        Returns:
+            New instance of the same class with complemented sequence.
+
+        Raises:
+            NotImplementedError: If _complement_map is not defined in subclass.
+        """
+        if self._complement_map is None:
+            raise NotImplementedError("Правила комплементарности не определены")
+        complemented = "".join([self._complement_map[c] for c in self.sequence])
+        return self.__class__(complemented)
+
+    def reverse(self):
+        """Return the reversed sequence."""
+        return self.__class__(self.sequence[::-1])
+
+    def reverse_complement(self):
+        """Return the reverse complement of the sequence."""
+        return self.complement().reverse()
+
+    def __len__(self):
+        return len(self.sequence)
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            return self.__class__(self.sequence[index])
+        else:
+            return self.sequence[index]
```

Comment:
А почему не просто вот так?
```suggestion
    def __getitem__(self, index):
        return self.__class__(self.sequence[index])
```